### PR TITLE
refactor: use synchronous message exchange for landscape-config

### PR DESCRIPTION
**Pull request by Perfect5th**

### CODE REVIEW

**Comment by Perfect5th** [comment]
This should more-or-less be a drop-in refactor, main differences being that registration does not depend on `landscape-client.service` starting, though it is still started after registration succeeds

To test, try `sudo ./scripts/landscape-config`

File: landscape/lib/compat.py
Comment (wck0) [inline landscape/lib/compat.py:1]: Can we add a task to remove this file?

**Review by wck0 — APPROVED** [review]
Manual testing went well.

LGTM

File: landscape/lib/compat.py
Comment (Perfect5th) [reply landscape/lib/compat.py:1]: Will do.

**Pull request merged**